apps/omg_eth/lib/omg_eth/encoding.ex

```diff
@@ -50,4 +50,39 @@ defmodule OMG.Eth.Encoding do
     {return, ""} = Integer.parse(encoded, 16)
     return
   end
+
+  @doc """
+  Encodes a list of smart contract constructor parameters into a base16 encoded-ABI that
+  solidity expects.
+  """
+  @spec encode_constructor_params(list()) :: String.t()
+  def encode_constructor_params(types_args) do
+    {types, args} = reduce_constructor_params(types_args)
+
+    args
+    |> ABI.TypeEncoder.encode_raw(types)
+    # NOTE: we're not using `to_hex` because the `0x` will be appended to the bytecode already
+    |> Base.encode16(case: :lower)
+  end
+
+  @doc """
+  Reduces a list of smart contract constructor parameters into a tuple of types list
+  and values list, so that it can be passed into `ABI.TypeEncoder.encode_raw/1`.
+  """
+  @spec reduce_constructor_params(list()) :: {list(), list()}
```

Comment:
maybe make this private and test via `encode...`. Looks like a worker function, it's only ever used in `encode` and tests

Reply:
I originally made it defp but it would make the tests less comprehensible so I don't know... :thinking:

Reply:
But actually.. I can decode then asserts. I'll update with your suggestions.

Reply:
If you want to test it separately move it to a separate module 😉

Reply:
@InoMurko Not sure if I get what you meant. Like this? https://github.com/omisego/elixir-omg/pull/1139/commits/29f6ffb2d45225a7e6a91ca68ad139c219bd171c

Reply:
bingo :)